### DISCUSSION TITLE
Loki: display formatted Context

### DISCUSSION THREAD

**Discussion by pgassmann** [grafana/grafana#40280] [General]
When the raw logs are json Lines, then they are very difficult to skim without previous formatting. With LogQL you can format the lines to a more human-friendly view.
In the explore view, the Context feature displays just the raw json lines with a lot of redundant information.

Request: Apply the formatting part of the LogQL also to the context display.
![image](https://user-images.githubusercontent.com/460192/136789789-1bf69fa4-3e51-4402-9903-93f6a00127a5.png)

Example Query:

```
{host=~"metrics1",category="dockerlogs",compose_project=~".*",compose_service=~".*"} |~ "" 
| json | line_format "{{.container_name}} {{ .source }} {{.message}}"
```
The formatting should also be applyed to the Context lines: `| json | line_format "{{.container_name}} {{ .source }} {{.message}}"`

**Comment by ivanahuckova** [comment]
@svennergr fyi

**Comment by MauroPfister** [comment]
I just came across this problem myself. It would be really awesome to have the context of a json log line formatted the same way as the line itself. Otherwise the context is not really useful at all. 

Is this somewhere on the roadmap?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Comment by avonar** [comment]
does anybody use this stack in production?
how do you research the problems without that feature?

it is not possible to read context, also it is not possible dive into context.
what is your happy path?

UPD: i think i got it, the happypath is to use `allow_structured_metadata` so the log from pod will be already parsed
